crates/containerd-shim-wasm/src/sys/unix/container/instance.rs

```
@@ -19,52 +20,50 @@ use crate::sys::container::executor::Executor;
 static DEFAULT_CONTAINER_ROOT_DIR: &str = "/run/containerd";
 
 pub struct Instance<E: Engine> {
-    engine: E,
     exit_code: ExitCode,
-    stdio: Stdio,
-    bundle: PathBuf,
     rootdir: PathBuf,
     id: String,
+    _phantom: PhantomData<E>,
 }
 
 impl<E: Engine> SandboxInstance for Instance<E> {
     type Engine = E;
 
-    fn new(id: String, cfg: Option<&InstanceConfig<Self::Engine>>) -> Self {
+    fn new(id: String, cfg: Option<&InstanceConfig<Self::Engine>>) -> Result<Self, SandboxError> {
         let cfg = cfg.unwrap();
         let engine = cfg.get_engine();
-        let bundle = cfg.get_bundle().unwrap_or_default().into();
+        let bundle = cfg.get_bundle().unwrap_or_default();
         let namespace = cfg.get_namespace();
         let rootdir = Path::new(DEFAULT_CONTAINER_ROOT_DIR).join(E::name());
         let rootdir = determine_rootdir(&bundle, &namespace, rootdir).unwrap();
```

since we return an error here, we can bubble this error up now?